### PR TITLE
ref #2: Fix AddTranslationsPass for Symfony 3+

### PR DESCRIPTION
**Pull request by umulmrum**
| Q             | A
| ------------- | ---
| Branch        | 6.2.x
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | #2
| License       | MIT

### CODE REVIEW

**Comment by UlrichKu** [comment]
Why must this be merged to 6.2? The issue talks about Symfony 3+?

**Comment by umulmrum** [comment]
@UlrichKu This bundle is independent of Chameleon and has been announcing compatibility with Symfony 3 and 4 for some time, but obviously we didn't test well enough. Therefore we should see compatibility as a fix here while it is a feature for Chameleon (also note the separate ticket for this PR).

File: DependencyInjection/Compiler/AddTranslationsPass.php [hunk 1]
```diff
@@ -27,9 +28,23 @@ public function process(ContainerBuilder $container)
     {
         $translatorDefinition = $container->getDefinition('translator.default');
 
+        $index = 3;
+        if (false === $this->hasResourceFilesOption($translatorDefinition, $index)) {
```
[the remark this answers is not in the document]
Comment (umulmrum) [reply DependencyInjection/Compiler/AddTranslationsPass.php]: We add translations manually to the Symfony translator and for this we need to find the argument that contains the relevant options. In Symfony 2 the index of the options argument differs from the index in Symfony 3, so we need to poke around until we got it.

Admittedly this whole compiler pass is a bit strange, but there is no way both "official" and easy that allows to add translations which are defined in a non-standard place (i.e. outside of Resources/translations).

I added some explanation in the method docblock - better?

[the remark this answers is not in the document]
Comment (UlrichKu) [reply DependencyInjection/Compiler/AddTranslationsPass.php]: Not here...

[the remark this answers is not in the document]
Comment (umulmrum) [reply DependencyInjection/Compiler/AddTranslationsPass.php]: But now :-)

**Comment by UlrichKu** [comment]
Pretty CI fails

**Comment by umulmrum** [comment]
Orrr -fixed :-)

**Pull request merged**